apps/settings/js/panels/developer_service_workers/developer_service_workers.js

```diff
@@ -0,0 +1,257 @@
+ *
+ *      Settings App
+ *          |
+ *          |- IAC
+ *          |
+ *      System App
+ *          |
+ *          |- Chrome/Content events
+ *          |
+ *        Gecko (ServiceWorkersManager accessed from a B2G dedicated component)
+ */
+
+define(function(require) {
+  'use strict';
+
+  /**
+   * @alias module:developer_service_workers/developer_service_workers
+   * @class DeveloperServiceWorkers
+   * @returns {DeveloperServiceWorkers}
+   */
+  var DeveloperServiceWorkers = function() {
+    this._elements = {};
+  };
+
+  DeveloperServiceWorkers.prototype = {
+    /**
+     * Initialization.
+     *
+     * @access public
+     * @memberOf DeveloperServiceWorkers.prototype
+     * @param  {Object} elements
+     */
+    init: function(elements) {
+      this._elements = elements;
+      this.refresh();
+      this.listeners = {};
+      this.serviceWorkersCount = 0;
+    },
+
+    renderServiceWorkerInfo: function(serviceWorkersInfo) {
+      var principal = serviceWorkersInfo.principal;
+      var scope = serviceWorkersInfo.scope;
+
+      var list = this._elements.serviceWorkersList;
+      list.innerHTML = '';
+
+      var fragment = document.createDocumentFragment();
+
+      var div = document.createElement('div');
+      div.dataset.scope = scope;
+
+      var header = document.createElement('header');
+      var headerContent = document.createElement('h2');
+      headerContent.textContent = principal.origin;
+      header.appendChild(headerContent);
+      div.appendChild(header);
+
+      [{
+        l10nId: 'service-worker-scope',
+        textContent: serviceWorkersInfo.scope
+      }, {
+        l10nId: 'service-worker-script-spec',
+        textContent: serviceWorkersInfo.scriptSpec
+      }, {
+        l10nId: 'service-worker-current-worker-url',
+        textContent: serviceWorkersInfo.currentWorkerURL
+      }, {
+        l10nId: 'service-worker-active-cache-name',
+        textContent: serviceWorkersInfo.activeCacheName
+      }, {
+        l10nId: 'service-worker-waiting-cache-name',
+        textContent: serviceWorkersInfo.waitingCacheName
+      }].forEach(property => {
+        var li = document.createElement('li');
+        var span = document.createElement('span');
+        span.dataset.l10nId = property.l10nId;
+        li.appendChild(span);
+        var small = document.createElement('small');
+        small.textContent = property.textContent;
+        li.appendChild(small);
+        div.appendChild(li);
+      });
+
+      // We store the listeners so we can remove them in case that
+      // we unregister the service worker.
+      if (!this.listeners[scope]) {
+        var self = this;
+        this.listeners[scope] = {
+          update: function() {
+            self.update(scope);
+          },
+          unregister: function() {
+            self.unregister(principal, scope);
+          }
+        };
+      }
+
+      var update = document.createElement('button');
```

Comment:
Let's wrap the update and unregister buttons in `li` elements in order to have correct styling.